test/Microsoft.TestPlatform.AcceptanceTests/RunsettingsTests.cs

```diff
@@ -573,63 +554,17 @@ private string GetRunsettingsFilePath(Dictionary<string, string> runConfiguratio
                 arguments = string.Concat(arguments, " -- ", runSettingsArgs);
             }
 
-            var cts = new CancellationTokenSource();
-            var numOfProcessCreatedTask = NumberOfProcessLaunchedUtility.NumberOfProcessCreated(
-                cts,
-                testhostProcessNames);
-
             this.InvokeVsTest(arguments);
-            cts.Cancel();
 
-            var processesCreated = numOfProcessCreatedTask.Result;
             // assert
-            Assert.AreEqual(
-                expectedNumOfProcessCreated,
-                processesCreated.Count,
-                $"Number of { string.Join(", ", testhostProcessNames) } process created, expected: {expectedNumOfProcessCreated} actual: {processesCreated.Count} ({ string.Join(", ", processesCreated) }) args: {arguments} runner path: {this.GetConsoleRunnerPath()}");
+            AssertExpectedNumberOfHostProcesses(expectedNumOfProcessCreated, workspace.Path, testhostProcessNames, arguments, this.GetConsoleRunnerPath());
             this.ValidateSummaryStatus(2, 2, 2);
 
             //cleanup
             if (!string.IsNullOrWhiteSpace(runsettingsPath))
             {
                 File.Delete(runsettingsPath);
             }
-            TryRemoveDirectory(resultsDir);
-        }
-
-        private int GetExpectedNumOfProcessCreatedForWithoutParallel()
```

Comment:
I don't know if I would remove this one...looks a good "piece of documentation"

Reply:
It's useful info but we are no longer looking at the processes to find how many we started, instead we ask each process to write down some logs and then we count them. Main difference is that we used to count some processes "wrongly" depending on whether they were run in process or not.